src/vss_tools/vspec/vssexporters/vss2samm/helpers/ttl_helper.py

```diff
@@ -0,0 +1,190 @@
+    node_uri = ttl_builder.add_graph_node(graph, vss_node, is_aspect)
+
+    if hasattr(vss_node.data, "type") and vss_node.data.type is NodeType.BRANCH:
+        node_char_uri = ttl_builder.add_node_branch_characteristic(graph, vss_node, node_uri)
+
+        if (
+            hasattr(vss_node.data, "instances")
+            and vss_node.data.instances
+            and vss_helper.is_node_expanded(vss_node) is False
+        ):
+            # Build instance(s) node(s) for the current vss_node ONLY when node is NOT EXPANDED.
+            # Otherwise, the expanded child nodes will cause conflicts with generated instance nodes
+            # and mess up the generated aspect model.
+            #
+            # NOTE: in this case the node's characteristic (node_char_uri),
+            #       will become a characteristic of each of this vss_node's instance(s)
+            node_name = get_node_name_from_vspec_uri(node_uri)
+
+            instances_dict_tree = vss_helper.get_instances_dict_tree(vss_node.data.instances, node_name)
+
+            node_instance_uri = ttl_builder.add_node_instances(graph, instances_dict_tree, node_char_uri)
+
+            # Link current vss_node to its instance uri, as if the instance uri is characteristic of this vss_node
+            graph.add((node_uri, SammConcepts.CHARACTERISTIC_RELATION.uri, node_instance_uri))
+
+        else:
+            # Link the characteristic uri to current vss_node's node_uri as usual
+            graph.add((node_uri, SammConcepts.CHARACTERISTIC_RELATION.uri, node_char_uri))
+
+        handle_branch_node(path_to_ttl, graph, vss_node, split_vss, node_uri, node_char_uri)
+
+    if vss_node.is_leaf:
+        ttl_builder.add_node_leaf(graph, node_uri, vss_node)
+
+    return node_uri
+
+
+def handle_branch_node(
+    path_to_ttl: Path, graph: Graph, vss_node: VSSNode, split_vss: bool, node_uri: URIRef, node_char_uri: URIRef
+):
+    log.debug("Handle branch node for VSSNode: '%s'", vss_node.name)
+
+    # Create node Entity to this vss_node branch
+    # NOTE: this will be like a Class representation of the current vss_node
+    #       In order to keep consistent the naming of semantic nodes,
+    #       we should append 'Entity' to the node's name, taken from its node_uri.
+    node_name = get_node_name_from_vspec_uri(node_uri)
+
+    # Node Entity name should be in camel case format with its first character in UPPER CASE
+    node_entity_name = str_to_uc_first_camel_case(node_name + "Entity")
+    node_entity_uri = URIRef(VSSConcepts.EMPTY.uri_string + node_entity_name)
+
+    # Add the node entity to the graph
+    graph.add((node_entity_uri, RDF.type, SammConcepts.ENTITY.uri))
+
+    # Add the node entity to its characteristic as data type
+    graph.add((node_char_uri, SammConcepts.DATA_TYPE.uri, node_entity_uri))
+
+    # Populate Entity properties if the current vss_node holds any child node
+    if hasattr(vss_node, "children") and len(vss_node.children) > 0:
```

Comment:
just `if vss_node.children`

Reply:
Done.
Removed this redundant check for children length.